daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/SAXInfosetInputter.scala

```diff
@@ -183,22 +193,23 @@ class SAXInfosetInputter(
 
   override protected def run(): Unit = {
     try {
-      // startDocument kicks off this entire process, so it should be on the queue so the
-      // waitForResume call can grab it. That is set to our current event, so when hasNext is called
-      // the nextEvent after the StartDocument can be queued
-      copyEvent(source = this.waitForResume(), dest = currentEvent)
+      // startDocument kicks off this entire process, so the first batch of events starting with it
+      // should be on the queue so the waitForResume call can grab it. This populates the
+      // batchedInfosetEvents global var for use by the Inputter
```

Comment:
Can't be "global" in the sense of a JVM-wide singleton here. It has to be per inputter. Otherwise you can't run multiple SAX parsers/unparsers at the same time in one JVM. We need to be able to do that. 

I assume this isn't really "global", so please update comment and indicate the real scope of this var, which is, I think a private var in this SAXInfosetInputter ??